lastseen.py

```diff
@@ -0,0 +1,46 @@
+import re
+from datetime import datetime
+from errbot import BotPlugin, botcmd, re_botcmd
+
+
+USAGE_STR = 'Nick not specified. (usage: !last_seen <nick>)'
+
+
+class LastSeen(BotPlugin):
+
+    def activate(self):
+        super(LastSeen, self).activate()
+        try:
+            if type(self['last_seen']) is not dict:
+                self['last_seen'] = {}
+        except KeyError:
+            self['last_seen'] = {}
+
+    @re_botcmd(pattern=r'^', prefixed=False, flags=re.IGNORECASE)
+    def update_last_seen(self, msg, match):
+        user = str(msg.frm).split('!')[0]
+        time = datetime.now()
+
+        last_seens = self['last_seen']
+        last_seens[user] = time
+        self['last_seen'] = last_seens
+
+    @botcmd
+    def last_seen(self, msg, args):
+        """Return last activity and its time of given user.
+
+        Examples:
+            !last_seen <user>
+        """
+        last_seens = self['last_seen']
+
+        if len(args) < 1:
+            return USAGE_STR
+        user = args.split(' ')[0]
+
+        try:
+            when = last_seens[user].strftime("%d-%m-%Y %H:%M")
```

Comment:
@JakubNvk I am wondering if the strftime's format here could be put into the [config](http://errbot.io/en/latest/user_guide/plugin_development/configuration.html) values? That would allow the admin to set a different one, if their cultural norms dictate another way of structuring the time information in ordinary situations.

Moreover, could the default be set to `"%d-%m-%Y %H:%M:%S"`?